### ISSUE TITLE
Auto lock should only happen when entering vim from normal mode

### ISSUE THREAD

**Issue by allpan3** [fresh2dev/zellij-autolock#5]
It seems like right now zellij will enter locked mode automatically whenever it enters vim no matter what the current mode is. For example, if I'm in tab mode and using h/l to switch tabs, it would abruptly enter locked mode whenever it happens to focus on a pane with vim running. That shouldn't be the expected behavior.

This may not be so simple to handle since if we indeed intend to stop at a vim pane, once we hit escape it will enter normal mode, but we expect it to be in locked mode. Not sure if the plugin could support this.

**Comment by fresh2dev** [comment]
Thanks for reporting this. It does seem like a tricky edge case. I don't use tab-mode; I use Alt h/l to switch tabs.

**Comment by allpan3** [comment]
The example I gave is minor. I also changed to use alt-h/l for tabs now.
There are some other cases that are somewhat more important. For example, if I'm in search mode in a pane, and I want to switch to another tab/pane to do some stuff and switch back. Normally, zellij would stay in search mode and the search patterns remain highlighted in the first pane. If vim was entered, it forces zellij to enter locked mode, therefore the search patterns are lost.

I think it makes sense to only enter locked mode from normal mode. Also would be good to check the front process when entering from another mode to normal mode. E.g. if I enter vim in search mode and decide to forgo the search mode and do work in vim by hitting escape to enter normal mode, it should automatically change it to locked mode.

Anyways thanks for the plugin. It works most smoothly among all the solutions I've tried so far. I'd like to contribute on this, only if I know some rust 😅.
